src/libraries/System.Security.Cryptography/tests/X509Certificates/ExportTests.cs

```diff
@@ -604,5 +754,124 @@ private static (byte[] Pkcs12, TKey key) CreateSimplePkcs12<TKey>() where TKey :
                 }
             }
         }
+
+        internal static (int certs, int keys) VerifyPkcs12(
+            byte[] pkcs12,
+            string password,
+            int expectedIterations,
+            HashAlgorithmName expectedMacHashAlgorithm,
+            PbeEncryptionAlgorithm expectedEncryptionAlgorithm)
+        {
+            const string Pkcs7Data = "1.2.840.113549.1.7.1";
+            const string Pkcs7Encrypted = "1.2.840.113549.1.7.6";
+
+            Pkcs12Info info = Pkcs12Info.Decode(pkcs12, out int read);
+            Assert.Equal(pkcs12.Length, read);
+            Assert.Equal(Pkcs12IntegrityMode.Password, info.IntegrityMode);
+            Assert.True(info.VerifyMac(password), nameof(info.VerifyMac));
+
+            PfxAsn pfxAsn = PfxAsn.Decode(pkcs12, AsnEncodingRules.BER);
```

Comment:
"Kevin... if you're using Pkcs12Info, why do you also need PfxAsn?"

-"Because Pkcs12Info says it _is_ encrypted, but not _how_ it was encrypted."

"Oh.  Maybe we should expose that instead?"

-(severe look of incredulity)

"Hmmm."

-(look continues)

"Good job, Kevin.  Smashed it."